pkg/util/kubernetes/kubelet/json.go

```diff
@@ -0,0 +1,99 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+// +build kubelet
+
+package kubelet
+
+import (
+	"time"
+	"unsafe"
+
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+)
+
+// jsoniterConfig mirrors jsoniter.ConfigFastest
+var jsonConfig = jsoniter.Config{
+	EscapeHTML:                    false,
+	MarshalFloatWith6Digits:       true,
+	ObjectFieldMustBeSimpleString: true,
+}
+
+// podUnmarshaller handles unmarshalling and filtering the podlist contents
+// according to the kubernetes_pod_expiration_minutes setting. It uses jsoniter
+// under the hood, with a custom decoder.
+type podUnmarshaller struct {
+	jsonConfig            jsoniter.API
+	podExpirationDuration time.Duration
+	timeNowFunction       func() time.Time // Allows to mock time in tests
+}
+
+func newPodUnmarshaller() *podUnmarshaller {
+	pu := &podUnmarshaller{
+		podExpirationDuration: config.Datadog.GetDuration("kubernetes_pod_expiration_minutes") * time.Minute,
+		timeNowFunction:       time.Now,
+	}
+
+	if pu.podExpirationDuration > 0 {
+		jsoniter.RegisterTypeDecoderFunc("kubelet.PodList", pu.filteringDecoder)
+	} else {
+		// Force-unregister for unit tests to pick up the right state
+		jsoniter.RegisterTypeDecoder("kubelet.PodList", nil)
+	}
+
+	// Build a new frozen config to invalidate type decoder cache
+	pu.jsonConfig = jsonConfig.Froze()
+
+	return pu
+}
+
+// unmarshal is a drop-in replacement for json.Unmarshall
+func (pu *podUnmarshaller) unmarshal(data []byte, v interface{}) error {
+	return pu.jsonConfig.Unmarshal(data, v)
+}
+
+func (pu *podUnmarshaller) filteringDecoder(ptr unsafe.Pointer, iter *jsoniter.Iterator) {
+	p := (*PodList)(ptr)
+	cutoffTime := pu.timeNowFunction().Add(-1 * pu.podExpirationDuration)
+
+	podCallback := func(iter *jsoniter.Iterator) bool {
+		pod := &Pod{}
+		iter.ReadVal(pod)
+
+		// Quick exit for running/pending containers
+		if pod.Status.Phase == "Running" || pod.Status.Phase == "Pending" {
```

Comment:
You can use: `k8s.io/api/core/v1/PodRunning|PodPending`  https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2384